cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java

```diff
@@ -583,22 +588,70 @@ void setReplicaLoad(TopicPartition tp, AggregatedMetricValues aggregatedMetricVa
     return brokerMap;
   }
 
+  /**
+   * Get per-logdir disk utilization information of the broker.
+   *
+   * @return The per-logdir disk utilization information. If information of replica placement over disk is not populated,
+   *         return null.
+   */
+  public Map<String, Double> diskUtils() {
+    if (_diskByLogdir.isEmpty()) {
+      return null;
+    }
+    if (_diskByLogdir.values().stream().mapToInt(disk -> disk.replicas().size()).sum() == 0) {
+      return null;
+    }
```

Comment:
Instead of using `_diskByLogdir.values().stream().mapToInt(disk -> disk.replicas().size()).sum() == 0` to check either replica-over-logdir information is populated or not(because it is unable to distinguish `disk with no replica` and `not populate replica information on disk`), I add a new flag to indicate explicitly indicate we populate or not.

Reply:
It is not clear why we would need `_replicaPlacementInfoPopulated` check here -- i.e. even if we don't have that check, we return an empty map either way because if `_replicaPlacementInfoPopulated` is `false`, then `_diskByLogdir` is empty, am I missing something?